lib/SILOptimizer/Transforms/SILMem2Reg.cpp

```diff
@@ -792,8 +790,11 @@ void StackAllocationPromoter::fixBranchesAndUses(BlockSet &PhiBlocks) {
     if (!BB->isEntry()) {
       if (auto *SI = dyn_cast<StoreInst>(Inst)) {
         if (SI->getOwnershipQualifier() == StoreOwnershipQualifier::Assign) {
-          SILValue Def = getLiveInValue(PhiBlocks, BB);
-          SILBuilderWithScope(SI).createDestroyValue(SI->getLoc(), Def);
+          if (destroyedBB.find(BB) == destroyedBB.end()) {
```

Comment:
You can use the `contains` method on `SmallPtrSet` here.

Reply:
thanks, fixed.